youtube_dl/extractor/iqiyi.py

```diff
@@ -301,10 +302,14 @@ def get_raw_data(self, tvid, video_id):
     def _extract_playlist(self, webpage):
         PAGE_SIZE = 50
 
-        links = re.findall(
-            r'<a[^>]+class="site-piclist_pic_link"[^>]+href="(http://www\.iqiyi\.com/.+\.html)"',
-            webpage)
-        if not links:
+        links = []
+        for link in re.findall(r'<a[^>]+class="[^"]*site-piclist_pic_link[^"]*"[^>]*>', webpage):
```

Comment:
@yan12125 Okay!

My suggestion is that we incoperate the changes in this pull request so that the Extractor is working again and we can close the issue, even though it's not using the new function get_elements_by_class.

And afterwards I'm building the "new" functions as you suggested and making all needed changes in a new pull request.